core/sr-primitives/src/traits.rs

```diff
@@ -966,10 +965,27 @@ pub trait RuntimeApiInfo {
 /// the transaction for the transaction pool.
 /// During block execution phase one need to perform the same checks anyway,
 /// since this function is not being called.
+#[deprecated(note = "Use SignedExtensions instead.")]
```

Comment:
How should we use `SignedExtensions` for this? You mean we should create an extra signed extension for each unsigned transaction and add this to the signed extensions used by the `UncheckedExtrinsic`? 

This would mean that we accept any unsigned transaction, when we forget to add the appropriate signed extension?

Reply:
Yes. It's a fair point that we should only allow unsigned transactions that are coming from one of the whitelisted modules (currently the `validate_unsigned` function of `SignedExtension` will just allow everything). But it should be easy to do a wrapper type that simply disallows a transaction if it's unsigned and none of the modules returns non-default `TransactionValidity`.

Actually it might be best to just return `InvalidTransaction` by default from `validate_unsigned` when we get rid of `UnsignedValidator`.

Reply:
Yeah, good ideas :)

Reply:
maybe one of our meta-modules (system etc.) can register a unified `CheckAndHandleUnsigned` signed extension. What this does is it rejects everything except for a handful ones that are explicitly whitelisted. This can be injected into system via `type UnsignedValidatorModules` which would be a tuple of modules that implement some trait (maybe even the current `ValidateUnsigned`)

This basically reverses the problem Basti said 

> This would mean that we accept any unsigned transaction, when we forget to add the appropriate signed extension?

everything is rejected unless a module explicitly ping system module about it.

Reply:
Okay what I said ^^ is in the same line of: 

> But it should be easy to do a wrapper type that simply disallows a transaction if it's unsigned and none of the modules returns non-default TransactionValidity.

i just understood it :D